cmake/modules/extensions.cmake

```diff
@@ -2487,6 +2487,63 @@ function(zephyr_file_copy oldname newname)
   endif()
 endfunction()
 
+# Usage:
+#   zephyr_file_suffix(<filename>)
+#
+# Zephyr file add suffix extension.
+# This function will check the provied filename or list of filenames to see if they have a
+# `_<FILE_SUFFIX>` extension to them and if so, updates the supplied variable/list with the new
+# path/paths.
+#
+# <filename>: Variable or list of absolute path filename(s) which should be checked and updated if
+#             there is a filename which has the FILE_SUFFIX present.
+#
+# Returns an updated variable/list of absolute path(s)
+#
+function(zephyr_file_suffix filename)
+  if(NOT DEFINED FILE_SUFFIX)
+    # If the file suffix variable is not known then there is nothing to do, return early
+    return()
+  endif()
+
+  # Check if the user supplied a list or a single item
+  list(LENGTH ${filename} list_length)
+
+  if(list_length GREATER 1)
+    set(tmp_list ${${filename}})
+    set(tmp_new_list)
+
+    # Loop for each list element
+    while(list_length GREATER 0)
+      list(POP_FRONT tmp_list tmp_element)
+      zephyr_file_suffix(tmp_element)
```

Comment:
This doesn't need to be recursive. A single item should be handled as a list of length 1.